assets/dns-app/resources/dns.yaml

```diff
@@ -289,3 +305,685 @@ spec:
+        [ 2656, 167 ],
+        [ 2672, 168 ],
+        [ 2688, 169 ],
+        [ 2704, 170 ],
+        [ 2720, 171 ],
+        [ 2736, 172 ],
+        [ 2752, 173 ],
+        [ 2768, 174 ],
+        [ 2784, 175 ],
+        [ 2800, 176 ],
+        [ 2816, 177 ],
+        [ 2832, 178 ],
+        [ 2848, 179 ],
+        [ 2864, 180 ],
+        [ 2880, 181 ],
+        [ 2896, 182 ],
+        [ 2912, 183 ],
+        [ 2928, 184 ],
+        [ 2944, 185 ],
+        [ 2960, 186 ],
+        [ 2976, 187 ],
+        [ 2992, 188 ],
+        [ 3008, 189 ],
+        [ 3024, 190 ],
+        [ 3040, 191 ],
+        [ 3056, 192 ],
+        [ 3072, 193 ],
+        [ 3088, 194 ],
+        [ 3104, 195 ],
+        [ 3120, 196 ],
+        [ 3136, 197 ],
+        [ 3152, 198 ],
+        [ 3168, 199 ],
+        [ 3184, 200 ],
+        [ 3200, 201 ],
+        [ 3216, 202 ],
+        [ 3232, 203 ],
+        [ 3248, 204 ],
+        [ 3264, 205 ],
+        [ 3280, 206 ],
+        [ 3296, 207 ],
+        [ 3312, 208 ],
+        [ 3328, 209 ],
+        [ 3344, 210 ],
+        [ 3360, 211 ],
+        [ 3376, 212 ],
+        [ 3392, 213 ],
+        [ 3408, 214 ],
+        [ 3424, 215 ],
+        [ 3440, 216 ],
+        [ 3456, 217 ],
+        [ 3472, 218 ],
+        [ 3488, 219 ],
+        [ 3504, 220 ],
+        [ 3520, 221 ],
+        [ 3536, 222 ],
+        [ 3552, 223 ],
+        [ 3568, 224 ],
+        [ 3584, 225 ],
+        [ 3600, 226 ],
+        [ 3616, 227 ],
+        [ 3632, 228 ],
+        [ 3648, 229 ],
+        [ 3664, 230 ],
+        [ 3680, 231 ],
+        [ 3696, 232 ],
+        [ 3712, 233 ],
+        [ 3728, 234 ],
+        [ 3744, 235 ],
+        [ 3760, 236 ],
+        [ 3776, 237 ],
+        [ 3792, 238 ],
+        [ 3808, 239 ],
+        [ 3824, 240 ],
+        [ 3840, 241 ],
+        [ 3856, 242 ],
+        [ 3872, 243 ],
+        [ 3888, 244 ],
+        [ 3904, 245 ],
+        [ 3920, 246 ],
+        [ 3936, 247 ],
+        [ 3952, 248 ],
+        [ 3968, 249 ],
+        [ 3984, 250 ],
+        [ 4000, 251 ],
+        [ 4016, 252 ],
+        [ 4032, 253 ],
+        [ 4048, 254 ],
+        [ 4064, 255 ],
+        [ 4080, 256 ]
+      ]
+    }
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: autoscaler-coredns-worker
+  namespace: kube-system
+  labels:
+    k8s-app: autoscaler-coredns-worker
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      k8s-app: autoscaler-coredns-worker
+  template:
+    metadata:
+      labels:
+        k8s-app: autoscaler-coredns-worker
+    spec:
+      serviceAccountName: cluster-proportional-autoscaler-coredns
+      tolerations:
+        - operator: "Exists"
+      containers:
+        - image: k8s.gcr.io/cpa/cluster-proportional-autoscaler-amd64:1.8.3
+          name: autoscaler
+          command:
+            - /cluster-proportional-autoscaler
+            - --namespace=kube-system
+            - --configmap=autoscaler-coredns-worker
+            - --target=deployment/coredns-worker
```

Comment:
So I'm curious, do you know what the usual deployment model with this autoscaler is? I.e. if we want to use it for some other component (e.g. prometheus in monitoring-app), would we need to deploy another instance of autoscaler specifically for monitoring, or reuse this one (like, add another config or something).

If it's supposed to be reused, I'm wondering if it would make sense instead of bundling it with dns make it a separate system level application ("autoscaler-app") and have specific apps that need its functionality just create their configurations.

Reply:
AFAIK we would need to deploy another instance... that targets the specific deployment. So it makes sense to bundle the autoscaler with the configuration its scaling. IE. If I wanted to scale both workers and masters, I would need to deploy 2x auto scalers targeting each deployment individually.

This is why I included it within coredns, and means we should be careful to bump in multiple repos... although it's not the end of the world if we get into a situation where we have multiple versions.